tests/integration/test_charm_with_proxy.py

```diff
@@ -40,26 +52,87 @@ async def proxy_fixture() -> AsyncIterator[str]:
 
 
 @pytest_asyncio.fixture(scope="module", name="app_with_aproxy")
 async def app_with_aproxy_fixture(
-    model: Model, app_no_runner: Application, proxy: str
+    model: Model,
+    charm_file: str,
+    app_name: str,
+    path: str,
+    token: str,
+    proxy: str,
 ) -> Application:
-    """Application configured to use aproxy"""
-
+    """Application with aproxy setup and firewall to block all other network access."""
     await model.set_config(
         {
+            "apt-http-proxy": proxy,
+            "apt-https-proxy": proxy,
+            "apt-no-proxy": "127.0.0.1,localhost,::1",
             "juju-http-proxy": proxy,
             "juju-https-proxy": proxy,
-            "juju-no-proxy": "",
+            "juju-no-proxy": "127.0.0.1,localhost,::1",
+            "snap-http-proxy": proxy,
+            "snap-https-proxy": proxy,
+            "snap-no-proxy": "127.0.0.1,localhost,::1",
             "logging-config": "<root>=INFO;unit=DEBUG",
         }
     )
 
-    await app_no_runner.set_config({"experimental-use-aproxy": "true"})
+    machine = await model.add_machine(constraints={"root-disk": 15}, series="jammy")
+    # Wait until juju agent has the hostname of the machine.
+    for _ in range(120):
+        if machine.hostname is not None:
+            break
+        await sleep(10)
+    else:
+        assert False, "Timeout waiting for machine to start"
+
+    # Disable external network access for the juju machine.
+    proxy_url = urlparse(proxy)
+    await machine.ssh(f"sudo iptables -I OUTPUT -d {proxy_url.hostname} -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 0.0.0.0/8 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 10.0.0.0/8 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 100.64.0.0/10 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 127.0.0.0/8 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 169.254.0.0/16 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 172.16.0.0/12 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 192.0.0.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 192.0.2.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 192.88.99.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 192.168.0.0/16 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 198.18.0.0/15 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 198.51.100.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 203.0.113.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 224.0.0.0/4 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 233.252.0.0/24 -j ACCEPT")
+    await machine.ssh("sudo iptables -I OUTPUT -d 240.0.0.0/4 -j ACCEPT")
+    await machine.ssh("sudo iptables -P OUTPUT DROP")
+    # Test the external network access is disabled.
+    await machine.ssh("ping -c1 canonical.com 2>&1 | grep '100% packet loss'")
```

Comment:
I'm wondering if the output for this should be `asserted`?

Reply:
The `machine.ssh` will throw exception on non-zero exit code.
See `if process.returncode != 0:` in:
https://pythonlibjuju.readthedocs.io/en/latest/_modules/juju/machine.html#Machine.ssh

This is why the assertion logic is within the `ping -c1 canonical.com 2>&1 | grep '100% packet loss'` command.